astropy/stats/bayesian_blocks.py

```diff
@@ -286,24 +277,22 @@ def p0_prior(self, N, Ntot):
         """
         return 4 - np.log(73.53 * self.p0 * (N ** -0.478))
 
-    def gamma_prior(self, N, Ntot):
-        """
-        Basic prior, parametrized by slope gamma.
-        See eq. 3 in Scargle (2012)
-        """
-        if self.gamma == 1:
-            return 0
-        else:
-            return (np.log(1 - self.gamma)
-                    - np.log(1 - self.gamma ** (Ntot + 1))
-                    + N * np.log(self.gamma))
-
     # the fitness_args property will return the list of arguments accepted by
     # the method fitness().  This allows more efficient computation below.
     @property
     def _fitness_args(self):
         return signature(self.fitness).parameters.keys()
 
+    def compute_ncp_prior(self, N):
+        """
+        If ``ncp_prior`` is not explicitly defined, compute it from ``gamma``
+        or ``p0``.
+        """
+        if self.gamma is not None:
+            self.ncp_prior = -np.log(self.gamma)
```

Comment:
Those are some interesting ideas.

As for saving the number of blocks, I figured that it could be another array of size N that is saved alongside ``best`` and ``last`` that just stores then number of blocks for the best partition for a given "R". It shouldn't be too difficult to implement.